iota/types.py

```diff
@@ -60,6 +60,12 @@ def random(cls, length):
     alphabet  = list(itervalues(AsciiTrytesCodec.alphabet))
     generator = SystemRandom()
 
+    try:
+      length = length or cls.LEN
+    except AttributeError:  # class has no LEN attribute
+      if length is None:
+        raise TypeError("random() missing 1 required positional argument: 'length'")
```

Comment:
I feel like this might be confusing to some users, since the method signature shows `length` as optional.

Maybe we could make this error explain that `{class}` does not define a length property.